buffer/src/main/java/io/netty/buffer/AbstractByteBuf.java

```diff
@@ -887,9 +899,11 @@ public ByteBuf readBytes(ByteBuf dst) {
 
     @Override
     public ByteBuf readBytes(ByteBuf dst, int length) {
-        if (length > dst.writableBytes()) {
-            throw new IndexOutOfBoundsException(String.format(
-                    "length(%d) exceeds dst.writableBytes(%d) where dst is: %s", length, dst.writableBytes(), dst));
+        if (checkBounds) {
+            if (length > dst.writableBytes()) {
```

Comment:
merge branches

Reply:
I've not merged this one because I don't want to loose the distinctions of the `dst.writableBytes` and `src.readableBytes` message logs in favour of something more generic to replace both.
If you want me to merge them please suggest me a proper meaningful message to help users when the checks will fail :+1: